tests/python/test_gtil.py

```diff
@@ -517,3 +517,167 @@ def test_lightgbm_sparse_categorical_model():
     Xa[Xa == 0] = "nan"
     out_pred = treelite.gtil.predict(tl_model, Xa, pred_margin=True)
     np.testing.assert_almost_equal(out_pred, expected_pred, decimal=5)
+
+
+@given(
+    dataset=standard_regression_datasets(),
+    predict_type=sampled_from(["leaf_id", "score_per_tree"]),
+    sample_size=integers(min_value=1, max_value=150),
+)
+@settings(**standard_settings())
+def test_predict_special_with_regressor(dataset, predict_type, sample_size):
+    # pylint: disable=too-many-locals
+    """Test predict_leaf / predict_per_tree with XGBoost regressor"""
+    X, y = dataset
+    dtrain = xgb.DMatrix(X, label=y)
+    assume(sample_size < X.shape[0])
+    param = {
+        "max_depth": 8,
+        "eta": 1,
+        "verbosity": 0,
+        "objective": "reg:squarederror",
+        "base_score": 0,
+    }
+    num_round = 10
+    xgb_model = xgb.train(
+        param,
+        dtrain,
+        num_boost_round=num_round,
+    )
+    model: treelite.Model = treelite.Model.from_xgboost(xgb_model)
+    assert model.num_tree == num_round
+
+    X_sample = X[0:sample_size]
+    if predict_type == "leaf_id":
+        leaf_pred = treelite.gtil.predict_leaf(model, X_sample)
+        assert leaf_pred.shape == (sample_size, num_round)
+        xgb_leaf_pred = xgb_model.predict(xgb.DMatrix(X_sample), pred_leaf=True)
+        assert np.array_equal(leaf_pred, xgb_leaf_pred)
+    else:
+        pred_per_tree = treelite.gtil.predict_per_tree(model, X_sample)
+        assert pred_per_tree.shape == (sample_size, num_round)
+        pred = xgb_model.predict(xgb.DMatrix(X_sample), output_margin=True)
+        np.testing.assert_almost_equal(np.sum(pred_per_tree, axis=1), pred, decimal=3)
+
+
+@given(
+    predict_type=sampled_from(["leaf_id", "score_per_tree"]),
+    random_seed=integers(min_value=0, max_value=50),
```

Comment:
I'm a bit confused about explicitly seeding via hypothesis, what is the motivation for this?

Reply:
It was to ensure that (X, y) data pair would come out randomly. Is it better to directly generate them from hypothesis strategies?

Reply:
I changed the test so that hypothesis directly generates the data matrix.